src/js/views.js

```diff
@@ -52,6 +71,51 @@ window.MapView = Backbone.View.extend({
             this.map.invalidateSize(animate);
         }
     },
+
+    updateMarkers: function() {
+        var subregions = null;
+
+        if (this.model.get('selectedRegion')) {
+            subregions = this.model.get('selectedRegion').get('subregions');
+
+            if (subregions && subregions.length > 0) {
+                _.forEach(subregions, function(region) {
+                    if (region.mapCenter) {
+                        this.markers.addLayer(this.createMarker(region));
+                    }
+                }, this);
+            }
+
+            this.map.addLayer(this.markers);
+        } else {
+            this.map.removeLayer(this.markers);
+            this.markers.clearLayers();
+        }
+    },
+
+    createMarker: function(region) {
+        var offset = [0, 25]; // Marker is used as popup tip
+        var popupContents = new MarkerPopupView({ model: region });
+        var marker = L.marker(region.mapCenter, { icon: this.icon })
+            .bindPopup(popupContents.render().el, { offset: offset });
```

Comment:
Do we need to manually destroy the `MarkerPopupView` objects when the markers are removed or will they be garbage collected when the markers are destroyed?

Reply:
Good question. I removed them manually to be safe in 6a6d4d4.